src/integer_mod_q/mat_zq/get.rs

```diff
@@ -174,6 +176,42 @@ impl MatZq {
 
         entries
     }
+
+    /// Computes the lengths of a given vector of [`fmpz`] structs
+    /// considering the [`Modulus`](crate::integer_mod_q::Modulus) of `self`.
+    ///
+    /// Parameters:
+    /// - `entries_fmpz`: holds a vector of [`fmpz`] structs
+    /// of which the length is calculated
+    ///
+    /// Returns a [`Vec<fmpz>`] of lengths of [`Zq`] entries
+    /// considering the [`Modulus`](crate::integer_mod_q::Modulus) of `self`.
+    ///
+    /// # Example
+    /// ```compile_fail
+    /// use qfall_math::intger_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let mat = MatZq::from_str("[[1,2],[3,4]] mod 3").unwrap();
+    /// let fmpz_entries = mat.collect_entries();
+    ///
+    /// let entry_lengths = mat.collect_lengths(&fmpz_entries);
+    /// ```
+    pub(crate) fn collect_lengths(&self) -> Vec<fmpz> {
```

Comment:
Memory Management:
Make the function unsafe since additional conditions have to be checked by the programmer:

The programmer is required to check that the returned values are properly cleared. I would also like to see this in the doc test (comment and/or clearing).

Or

return a `MatZ`

Reply:
Returns now a Vec<Z>, which does not require additional memory management anymore